Last login: Mon Oct 16 16:02:13 on ttys000
Ninas-MacBook-Pro:~ Niner$ cd Desktop/DojoAssignments
Ninas-MacBook-Pro:DojoAssignments Niner$ source flaskEnv/bin/activate 
(flaskEnv) Ninas-MacBook-Pro:DojoAssignments Niner$ cd Python/Django
(flaskEnv) Ninas-MacBook-Pro:Django Niner$ ls
Amadon		TimeDisplay	main		users
NinjaGold	django-admin	multipleapps	wordsession
RandomWord	djangoapp	surveyform
(flaskEnv) Ninas-MacBook-Pro:Django Niner$ django-admin startproject ninjas
(flaskEnv) Ninas-MacBook-Pro:Django Niner$ ls
Amadon		TimeDisplay	main		surveyform
NinjaGold	django-admin	multipleapps	users
RandomWord	djangoapp	ninjas		wordsession
(flaskEnv) Ninas-MacBook-Pro:Django Niner$ cd ninjas
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ mkdir apps
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ cd apps
(flaskEnv) Ninas-MacBook-Pro:apps Niner$ touch __init__.py
(flaskEnv) Ninas-MacBook-Pro:apps Niner$ python ../manage.py startapp ninjas_app
(flaskEnv) Ninas-MacBook-Pro:apps Niner$ cd ..
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ ls
apps		manage.py	ninjas
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py runserver
Performing system checks...

System check identified some issues:

WARNINGS:
?: (urls.W001) Your URL pattern '^$' uses include with a regex ending with a '$'. Remove the dollar from the regex to avoid problems including URLs.

System check identified 1 issue (0 silenced).

You have 13 unapplied migration(s). Your project may not work properly until you apply the migrations for app(s): admin, auth, contenttypes, sessions.
Run 'python manage.py migrate' to apply them.

October 16, 2017 - 21:24:59
Django version 1.11.6, using settings 'ninjas.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
python manage.py makemigrations
^C(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py makemigrations
System check identified some issues:

WARNINGS:
?: (urls.W001) Your URL pattern '^$' uses include with a regex ending with a '$'. Remove the dollar from the regex to avoid problems including URLs.
No changes detected
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py makemigrations
No changes detected
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py migrate
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, sessions
Running migrations:
  Applying contenttypes.0001_initial... OK
  Applying auth.0001_initial... OK
  Applying admin.0001_initial... OK
  Applying admin.0002_logentry_remove_auto_add... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying auth.0007_alter_validators_add_error_messages... OK
  Applying auth.0008_alter_user_username_max_length... OK
  Applying sessions.0001_initial... OK
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py makemigrations
Migrations for 'ninjas_app':
  apps/ninjas_app/migrations/0001_initial.py
    - Create model Dojos
    - Create model Ninjas
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py migrate
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, ninjas_app, sessions
Running migrations:
  Applying ninjas_app.0001_initial... OK
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py shell
Python 2.7.10 (default, Jul 30 2016, 18:31:42) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> Dojos.objects.create(name="CodingDojo Silicon Valley", city="Mountain View", state="Ca")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'Dojos' is not defined
>>> Dojos.objects.create(name="CodingDojo Silicon Valley", city="Mountain View", state="Ca")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'Dojos' is not defined
>>> exit()
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py makemigrations
No changes detected
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py migrate
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, ninjas_app, sessions
Running migrations:
  No migrations to apply.
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py shell
Python 2.7.10 (default, Jul 30 2016, 18:31:42) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from apps.ninjas_app.models import *
>>> Dojos.objects.create(name="CodingDojo in Silicon Valley", city="Mountain View",state="Ca")
<Dojos: Dojos object>
>>> Dojos.objects.create(name="CodingDojo in Seattle", city="Seattle",state="WA")
<Dojos: Dojos object>
>>> Dojos.objects.create(name="CodingDojo in New York", city="New York",state="NY")
<Dojos: Dojos object>
>>> Ninjas.objects.create(first_name="Evan",last_name"Munz", dojos=1)
  File "<console>", line 1
    Ninjas.objects.create(first_name="Evan",last_name"Munz", dojos=1)
                                                          ^
SyntaxError: invalid syntax
>>> Ninjas.objects.create(first_name="Evan",last_name="Munz", dojos=1)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Library/Python/2.7/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/django/db/models/query.py", line 392, in create
    obj = self.model(**kwargs)
  File "/Library/Python/2.7/site-packages/django/db/models/base.py", line 555, in __init__
    _setattr(self, field.name, rel_obj)
  File "/Library/Python/2.7/site-packages/django/db/models/fields/related_descriptors.py", line 216, in __set__
    self.field.remote_field.model._meta.object_name,
ValueError: Cannot assign "1": "Ninjas.dojos" must be a "Dojos" instance.
>>> Ninjas.objects.create(first_name="Evan",last_name"Munz", dojos=dojos.objects.get(id=1))
  File "<console>", line 1
    Ninjas.objects.create(first_name="Evan",last_name"Munz", dojos=dojos.objects.get(id=1))
                                                          ^
SyntaxError: invalid syntax
>>> Ninjas.objects.create(first_name="Evan",last_name"Munz", dojos=Dojos.objects.get(id=1))
  File "<console>", line 1
    Ninjas.objects.create(first_name="Evan",last_name"Munz", dojos=Dojos.objects.get(id=1))
                                                          ^
SyntaxError: invalid syntax
>>> Ninjas.objects.create(first_name="Evan",last_name"Munz", Dojos=dojos.objects.get(id=1))
  File "<console>", line 1
    Ninjas.objects.create(first_name="Evan",last_name"Munz", Dojos=dojos.objects.get(id=1))
                                                          ^
SyntaxError: invalid syntax
>>> Ninjas.objects.create(first_name="Evan",last_name"Munz", dojos=Dojos.objects.get(id=1))
  File "<console>", line 1
    Ninjas.objects.create(first_name="Evan",last_name"Munz", dojos=Dojos.objects.get(id=1))
                                                          ^
SyntaxError: invalid syntax
>>> exit()
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py migrate
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, ninjas_app, sessions
Running migrations:
  No migrations to apply.
  Your models have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py makemigrations
Did you rename ninjas.Dojos to ninjas.dojos (a ForeignKey)? [y/N]y            ^ Migrations for 'ninjas_app':
  apps/ninjas_app/migrations/0002_auto_20171016_2155.py
    - Rename field Dojos on ninjas to dojos
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py migrate
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, ninjas_app, sessions
Running migrations:
  Applying ninjas_app.0002_auto_20171016_2155... OK
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py shell
Python 2.7.10 (default, Jul 30 2016, 18:31:42) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from apps.ninjas_app.models import * 
>>> Ninjas.objects.create(first_name="Evan",last_name"Munz", dojos=Dojos.objects.get(id=1))
  File "<console>", line 1
    Ninjas.objects.create(first_name="Evan",last_name"Munz", dojos=Dojos.objects.get(id=1))
                                                          ^
SyntaxError: invalid syntax
>>> Ninjas.objects.create(first_name="Evan",last_name="Munz", dojos=Dojos.objects.get(id=1))
<Ninjas: Ninjas object>
>>> Ninjas.objects.create(first_name="Nina",last_name"Palumbo", dojos=Dojos.objects.get(id=1))
  File "<console>", line 1
    Ninjas.objects.create(first_name="Nina",last_name"Palumbo", dojos=Dojos.objects.get(id=1))
                                                             ^
SyntaxError: invalid syntax
>>> Ninjas.objects.create(first_name="Evan",last_name="Munz", dojos=Dojos.objects.get(id=1))
<Ninjas: Ninjas object>
>>> Ninjas.objects.create(first_name="Nina",last_name="Palumbo", dojos=Dojos.objects.get(id=1))
<Ninjas: Ninjas object>
>>> Ninjas.objects.create(first_name="Tom",last_name="Tom", dojos=Dojos.objects.get(id=2))
<Ninjas: Ninjas object>
>>> Ninjas.objects.create(first_name="Will",last_name="Will", dojos=Dojos.objects.get(id=2))
<Ninjas: Ninjas object>
>>> Ninjas.objects.create(first_name="Will",last_name="andrew", dojos=Dojos.objects.get(id=2))
<Ninjas: Ninjas object>
>>> Ninjas.objects.create(first_name="Kyle",last_name="Beatrice", dojos=Dojos.objects.get(id=3))
<Ninjas: Ninjas object>
>>> Ninjas.objects.create(first_name="Corey",last_name="adams", dojos=Dojos.objects.get(id=3))
<Ninjas: Ninjas object>
>>> Dojos.objects.get(id=1)
<Dojos: Dojos object>
>>> Dojos.objects.get("Ninjas")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Library/Python/2.7/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/django/db/models/query.py", line 371, in get
    clone = self.filter(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/django/db/models/query.py", line 784, in filter
    return self._filter_or_exclude(False, *args, **kwargs)
  File "/Library/Python/2.7/site-packages/django/db/models/query.py", line 802, in _filter_or_exclude
    clone.query.add_q(Q(*args, **kwargs))
  File "/Library/Python/2.7/site-packages/django/db/models/sql/query.py", line 1250, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
  File "/Library/Python/2.7/site-packages/django/db/models/sql/query.py", line 1276, in _add_q
    allow_joins=allow_joins, split_subq=split_subq,
  File "/Library/Python/2.7/site-packages/django/db/models/sql/query.py", line 1151, in build_filter
    arg, value = filter_expr
ValueError: too many values to unpack
>>> ninja.objects.filter(dojos_id=1).values()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'ninja' is not defined
>>> Ninjas.objects.filter(dojos_id=1).values()
<QuerySet [{u'dojos_id': 1, 'first_name': u'Evan', 'last_name': u'Munz', u'id': 1}, {u'dojos_id': 1, 'first_name': u'Evan', 'last_name': u'Munz', u'id': 2}, {u'dojos_id': 1, 'first_name': u'Nina', 'last_name': u'Palumbo', u'id': 3}]>
>>> Dojos.objects.first("Ninjas")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Library/Python/2.7/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
TypeError: first() takes exactly 1 argument (2 given)
>>> Dojos.objects.first().values()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
AttributeError: 'Dojos' object has no attribute 'values'
>>> Dojos.objects.first("name")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Library/Python/2.7/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
TypeError: first() takes exactly 1 argument (2 given)
>>> exit()
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py makemigrations
You are trying to add a non-nullable field 'desc' to dojos without a default; we can't do that (the database needs something to populate existing rows).
Please select a fix:
 1) Provide a one-off default now (will be set on all existing rows with a null value for this column)
 2) Quit, and let me add a default in models.py
Select an option: 1
Please enter the default value now, as valid Python
The datetime and django.utils.timezone modules are available, so you can do e.g. timezone.now
Type 'exit' to exit this prompt
>>> none
Invalid input: name 'none' is not defined
>>> "none"
Migrations for 'ninjas_app':
  apps/ninjas_app/migrations/0003_dojos_desc.py
    - Add field desc to dojos
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py shell
Python 2.7.10 (default, Jul 30 2016, 18:31:42) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from apps.ninjas_app.models import *
>>> Dojos.objects.all().values()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Library/Python/2.7/site-packages/django/db/models/query.py", line 226, in __repr__
    data = list(self[:REPR_OUTPUT_SIZE + 1])
  File "/Library/Python/2.7/site-packages/django/db/models/query.py", line 250, in __iter__
    self._fetch_all()
  File "/Library/Python/2.7/site-packages/django/db/models/query.py", line 1118, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/Library/Python/2.7/site-packages/django/db/models/query.py", line 106, in __iter__
    for row in compiler.results_iter():
  File "/Library/Python/2.7/site-packages/django/db/models/sql/compiler.py", line 836, in results_iter
    results = self.execute_sql(MULTI)
  File "/Library/Python/2.7/site-packages/django/db/models/sql/compiler.py", line 894, in execute_sql
    raise original_exception
OperationalError: no such column: ninjas_app_dojos.desc
>>> exit()
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py migrate
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, ninjas_app, sessions
Running migrations:
  Applying ninjas_app.0003_dojos_desc... OK
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ python manage.py shell
Python 2.7.10 (default, Jul 30 2016, 18:31:42) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from apps.ninjas_app.models import *
>>> Dojos.objects.all().values()
<QuerySet [{'city': u'Mountain View', 'state': u'Ca', 'desc': u'none', u'id': 1, 'name': u'CodingDojo in Silicon Valley'}, {'city': u'Seattle', 'state': u'WA', 'desc': u'none', u'id': 2, 'name': u'CodingDojo in Seattle'}, {'city': u'New York', 'state': u'NY', 'desc': u'none', u'id': 3, 'name': u'CodingDojo in New York'}]>
>>> exit()
(flaskEnv) Ninas-MacBook-Pro:ninjas Niner$ 
